### PR TITLE
Clean up policy actions

### PR DESCRIPTION
**Pull request by marcaaron**
cc @roryabraham Longer term we probably want to solve the issue of why we can't get policies and create them at the same time, but this feels like a logical refactor. I think by splitting these out into their own methods things feel more intentional - even if the behavior is exactly the same for now.

### Details
Clean up Policy actions

### Fixed Issues
$ https://github.com/Expensify/App/issues/6153

### Tests / QA Steps
1. Log out of NewDot
2. Sign up for a new account via OldDot on mobile web with a private domain
2. Choose "Set up my company for free"
3. Verify that the page opens to NewDot and the workspace is visible
4. Close the workspace and validate the email
5. Sign into OldDot with the same account (desktop web) and navigate to policies. Select the workspace.
6. Verify the page opens to NewDot with the workspace visible.
7. Sign out of NewDot and repeat steps 5 & 6

### Tested On

- [x] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

File: src/libs/actions/Policy.js [hunk 1]
```diff
@@ -144,40 +153,41 @@ function create(name = '', shouldAutomaticallyReroute = true) {
...
             return API.GetPolicyList();
         })
         .then((data) => {
             if (data.jsonCode === 200) {
-                const policyDataToStore = transformPolicyListToOnyxCollection(data.policyList || []);
-                updateAllPolicies(policyDataToStore);
+                lodashMerge(policyCollection, transformPolicyListToOnyxCollection(data.policyList || []));
+            }
+        })
+        .finally(() => {
```
Comment (roryabraham) [inline src/libs/actions/Policy.js:172]: Do we need a separate `finally` block here? Why not just move `updateAllPolicies(policyCollection)` into the if block of the previous `.then`?

**Review by roryabraham — DISMISSED** [review]

File: src/libs/actions/Policy.js (hunk 1 above)
Comment (marcaaron) [reply src/libs/actions/Policy.js:172]: I don't know if I'd say "need", but it seems fine. The main difference is that if we hit a `catch()` by using `finally()` then whatever data we successfully managed to merged into `policyCollection` will be set regardless. If we did it in the previous `.then()` and the second API throws then no data will be set.

File: src/pages/home/sidebar/SidebarScreen.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ import {
 } from '../../../components/Icon/Expensicons';
 import Permissions from '../../../libs/Permissions';
 import ONYXKEYS from '../../../ONYXKEYS';
-import {create, isAdminOfFreePolicy} from '../../../libs/actions/Policy';
+import * as Policy from '../../../libs/actions/Policy';
```
Comment (deetergp) [inline src/pages/home/sidebar/SidebarScreen.js:30]: NAB: Why `*` and not just replace `create` with `createAndNavigate`?

**Review by deetergp — APPROVED** [review]
Looks good and tests well. Had one NAB question that I'll wait on before merging.

File: src/pages/home/sidebar/SidebarScreen.js (hunk 2 above)
Comment (marcaaron) [reply src/pages/home/sidebar/SidebarScreen.js:30]: More context here: https://github.com/Expensify/App/issues/5984#issuecomment-952302958

Basically we are going to change these all eventually so I got to this one now.

**Pull request merged**